SearchParser.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * SearchParser
+ * @param {Array} patterns  Enumerates params and the patterns that match them
+ * @param {boolean} options.returnAllMatches  If true return all matches (not just first)
```

Comment:
I mean that whatever mySearchParser.match could pick the first element off
the object of matches. But since it returns an object, "first element" is
not well defined, so maybe this wouldn't work as I thought.

If you think returnAllMatches is an option that will receive minimal use,
you might consider removing it until it is needed.

(Not a blocking comment)

On Fri, Apr 15, 2016 at 16:57 Graham Batzler <notifications@github.com>
wrote:

> In SearchParser.js
> <https://github.com/batzlerg/search-parser/pull/1#discussion_r59939748>:
>
> > @@ -0,0 +1,49 @@
> > +/**
> > + * SearchParser
> > + * @param {Array} patterns  Enumerates params and the patterns that match them
> > + * @param {boolean} options.returnAllMatches  If true return all matches (not just first)
>
> returnAllMatches was more important when I was defining the schema as an
> object, since the order of properties couldn't be guaranteed.
>
> By 'the job of the calling function' do you mean it should return all
> matches by default?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/batzlerg/search-parser/pull/1/files/d10fbdc99b20a2b40002ad95a638553034b4f52e#r59939748>
>